server/api.go

```diff
@@ -59,8 +59,8 @@ func (h *handler) downloadDumpHandler(w http.ResponseWriter, r *http.Request) {
 	appCfg := h.plugin.API.GetConfig()
 	metricsFrom, ok := appCfg.PluginSettings.Plugins[root.Manifest.Id]["collect_metrics_from"]
 	if !ok {
-		w.WriteHeader(http.StatusInternalServerError)
-		return
+		// set it to the default value if the setting is not there
+		metricsFrom = "yesterday"
```

Comment:
Can we not check this in `.isValid` and if it's null, then set the value accordingly and write the config? That way there are less implicit assumptions in the code.